pactl-widget/pactl.lua

```diff
@@ -0,0 +1,124 @@
+local spawn = require("awful.spawn")
+local utils = require("awesome-wm-widgets.pactl-widget.utils")
+
+local pactl = {}
+
+
+function pactl.volume_increase(device, step)
+    spawn('pactl set-sink-volume ' .. device .. ' +' .. step .. '%', false)
+end
+
+function pactl.volume_decrease(device, step)
+    spawn('pactl set-sink-volume ' .. device .. ' -' .. step .. '%', false)
+end
+
+function pactl.mute_toggle(device)
+    spawn('pactl set-sink-mute ' .. device .. ' toggle', false)
+end
+
+function pactl.get_volume(device)
+    local stdout = utils.popen_and_return('pactl get-sink-volume ' .. device)
+
+    local volsum, volcnt = 0, 0
+    for vol in string.gmatch(stdout, "(%d?%d?%d)%%") do
+        vol = tonumber(vol)
+        if vol ~= nil then
+            volsum = volsum + vol
+            volcnt = volcnt + 1
+        end
+    end
+
+    if volcnt == 0 then
+        return nil
+    end
+
+    return volsum / volcnt
+end
+
+function pactl.get_mute(device)
+    local stdout = utils.popen_and_return('pactl get-sink-mute ' .. device)
+    if string.find(stdout, "yes") then
+        return true
+    else
+        return false
+    end
+end
+
+function pactl.get_sinks_and_sources()
+    local default_sink = utils.trim(utils.popen_and_return('pactl get-default-sink'))
+    local default_source = utils.trim(utils.popen_and_return('pactl get-default-source'))
+
+    local sinks = {}
+    local sources = {}
+
+    local device
+    local ports
+    local key
+    local value
+    local in_section
+
+    for line in utils.popen_and_return('pactl list'):gmatch('[^\r\n]*') do
+
+        if string.match(line, '^%a+ #') then
+            in_section = nil
+        end
+
+        is_sink_line = string.match(line, '^Sink #')
+        is_source_line = string.match(line, '^Source #')
```

Comment:
Could you mark them `local`, as lua-check [complains](https://github.com/streetturtle/awesome-wm-widgets/actions/runs/3841439437/jobs/6752164741)

Reply:
Done!